src/metabase/async/streaming_response.clj

```diff
@@ -236,7 +235,7 @@
                                 streaming-response))
            :headers      (cond-> (assoc headers "Content-Type" content-type)
                            gzip? (assoc "Content-Encoding" "gzip"))
-           :status       202)))
+           :status       (or (:status options) 202))))
```

Comment:
using the status passed from `stream-options` ([code](https://github.com/metabase/metabase/blob/c58a294548b5c05e924fb9b157182f9f0bd58493/src/metabase/query_processor/streaming/json.clj#L21))